### DISCUSSION TITLE
Antiflood/Antispam feature

### DISCUSSION THREAD

**Discussion by ghost** [nutgram/nutgram#258] [Ideas]
In Telegram, there are CLI bots that provide more features than API bots. In fact, CLI bots are for Telegram accounts. That is, you can write codes that make normal accounts in Telegram have more features! (act like a bot)

Telegram has limitations called floodwait for sending requests in bot API, etc. in some sections, but there are no restrictions for Telegram users who use bots!

This means that sometimes some users spam the API bot (in Telegram) in order to destroy it with CLI bots and cause the bot to malfunction (by creating many and consecutive requests).

If a system is designed to prevent these spams or just reject the requests of these types of users quickly, it will help the bots a lot. For example, design a system where each user can send only 2 requests per second, or in another example, each user can send only 3 requests in 5 seconds. If it exceeds this value, the bot will reject the requests sent by the user for, for example, 15 seconds

You may think that this is not necessary, and with the server upgrade, there will be no need for anti-spam and similar things. You are wrong! According to my personal experience, some users spam in a group (perhaps 50 users at the same time) in a part of the bot that requires a little more processing, and I also think that buying and upgrading a server in these cases is not a rational thing and just a waste. It's a cost, because we don't really need it, and only a few malicious users create a lot of requests

I hope Nutgram will add this feature in the next updates

Thank you

**Comment by Lukasss93** [comment]
I don't think this feature will be added in the near future but you can create it in your bot :)

See an example of this implementation in my bot (Laravel):
https://github.com/Lukasss93/telegram-stickeroptimizer/blob/master/app/Telegram/Middleware/CheckRateLimit.php

**Comment by sergix44** [comment]
This is not going to be added. 
As @Lukasss93 already replied, its the application that should define a rate limit strategy, as it may differ from a bot to another, and in high traffic bots it may be also harmful to the user experience.
